V1.2.0 rc

ecurley-nextcentury opened the pull request:
This includes all changes required for the M36 dry run and evaluation. Once this is in master, I will create the following artifacts:
* Packaged java release on OSSRH
* Validation docker image on github
* Github release
* (possibly) Packaged python release on pypi

ecurley-nextcentury commented:
@gitnordt, I had to bump the version to 1.2.1 due to that erroneous submission I made to OSSRH a couple weeks ago

gitnordt commented on docker/aif-validator/README.md:
```diff
@@ -0,0 +1,78 @@
+| :---------------------- | :------------------------------------------------------------- | :------------------- |
+| `VALIDATION_HOME`       | Location in the container for the built, git clone of AIF      | `/opt/aif-validator` |
+| `VALIDATION_FLAGS`      | Flags sent to the AIF validator (see below)                    | none                 |
+| `VALIDATION_LOG`        | `stdout` or a file in the mounted directory to save the log    | `stdout`             |
```
The file has '/v/validate_log.out' as the default. Please update here or in the run.sh.example.

ecurley-nextcentury replied:
The defaults here are the defaults for the image, not for the script. Leaving.

gitnordt commented on docker/aif-validator/README.md:
```diff
@@ -0,0 +1,78 @@
+| `VALIDATION_FLAGS`      | Flags sent to the AIF validator (see below)                    | none                 |
+| `VALIDATION_LOG`        | `stdout` or a file in the mounted directory to save the log    | `stdout`             |
+| `VALIDATE_DIR_OR_FILES` | Either `directory` or `files`                                  | `directory`          |
+| `TARGET_TO_VALIDATE`    | The directory to validate OR the space separated list of files | `${pwd}`             |
```
The file has '/v' as the default. Please update here or in the run.sh.example.

gitnordt commented on docker/aif-validator/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+FROM amazonlinux:latest
+MAINTAINER craig.warsaw@caci.com
```
Should Craig's email address be changed to yours? I also saw Craig referenced in the AIDA-Interchange-Format/README.md.

ecurley-nextcentury replied:
Good catch. I've removed MAINTAINER as it's deprecated and I've changed README.md to use aif-support@nextcentury.com

ecurley-nextcentury commented on python/tests/scaling_test.py:
```diff
@@ -183,62 +181,183 @@ def write_to_file(self, testname):
+        "Time"]
 
 if __name__ == "__main__":
     ScalingTest().run_scaling_test()
```
Fixed

ecurley-nextcentury commented on java/src/main/resources/com/ncc/aif/restricted_aif.shacl:
```diff
@@ -241,3 +230,106 @@ aida:LinkAssertionShape
+aida:ClusterMembersShape
+    a sh:NodeShape ;
+    sh:targetClass aida:SameAsCluster;
+    sh:sparql aida:ClusterMembersSameAsBaseClass .
```
Fixed

ecurley-nextcentury commented on java/src/main/java/com/ncc/aif/util/AIFComparableNode.java:
```diff
@@ -0,0 +1,125 @@
+         }
+     }
+ }
+ 
```
Fixed

ecurley-nextcentury commented on java/src/main/java/com/ncc/aif/InstrumentedValidationEngine.java:
```diff
@@ -0,0 +1,174 @@
+		engine.setConfiguration(configuration);
+		return engine;
+	}
+}
```
Fixed

ecurley-nextcentury commented on python/README.md:
````diff
@@ -1,22 +1,80 @@
+
+```bash
+$ virtualenv -p python3 ~/.virtualenvs/aida-interchange-format
+$ source ~/.virtualenvs/aida-interchange-format/bin/activate
````
Good point. Removed.

ecurley-nextcentury commented on python/README.md:
````diff
@@ -1,22 +1,80 @@
+```bash
+$ virtualenv -p python3 ~/.virtualenvs/aida-interchange-format
+$ source ~/.virtualenvs/aida-interchange-format/bin/activate
+$ pip install aida-interchange
````
No. I'm assuming they will want the latest version. Leaving this.

ecurley-nextcentury commented on python/README.md:
````diff
@@ -1,22 +1,80 @@
 
 ```bash
-$ python3 setup.py install
+$ pip install aida-interchange==x.x.x
````
I've removed the versions altogether.

ecurley-nextcentury commented on docker/aif-validator/README.md:
```diff
@@ -0,0 +1,78 @@
+| Env Variable            | Description                                                    | Default              |
+| :---------------------- | :------------------------------------------------------------- | :------------------- |
+| `VALIDATION_HOME`       | Location in the container for the built, git clone of AIF      | `/opt/aif-validator` |
+| `VALIDATION_FLAGS`      | Flags sent to the AIF validator (see below)                    | none                 |
```
The defaults here are the defaults for the image, not for the script. Leaving.

gitnordt replied:
Ok, just wanted to make sure you were aware.

ecurley-nextcentury commented:
@gitnordt, I think I've addressed all your concerns, but I've also made some changes extraneous to your suggestions. They should probably be vetted as well. Let me know if you have any questions
